test/performance/performance.go

```diff
@@ -0,0 +1,333 @@
+			if err == nil {
+				found = true
+				break
+			}
+			klog.V(2).Info("Secret not found, waiting 5 seconds...")
+			time.Sleep(5 * time.Second)
+		}
+
+		if !found {
+			klog.Fatal("Error: Prometheus secret creation timed out")
+		}
+	}
+
+	secret, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'")
+	if err != nil {
+		klog.Exitf("Error getting Prometheus metrics secret: %s", err)
+	}
+
+	token, err := bash(fmt.Sprintf("echo $(oc get secret %s -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d", secret))
+	if err != nil {
+		klog.Exitf("Error getting API token from secret: %s", err)
+	}
+
+	thanos_host, err := bash("oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'")
+	if err != nil {
+		klog.Exitf("Error getting Thanos host: %s", err)
+	}
+
+	klog.Info("Setup complete! Starting the Config Policy Controller performance test :)")
+	table_data := []metricData{}
+
+	table_data = append(table_data, getMetrics(thanos_host, token, per_batch_sleep, 0))
+
+	for total_plcs := 0; total_plcs < n_total; total_plcs += n_per_batch {
+		start := time.Now()
+
+		// create a batch of policies
+		klog.V(2).Infof("Creating %d copies of %s on the managed cluster...\n", n_per_batch, plc_filename)
+		batch_fails := 0
+		for batch_plcs := 0; batch_plcs < n_per_batch; batch_plcs += 1 {
+			sed, err := bash(fmt.Sprintf(`sed "s/\[ID\]/%s/" %s > %s/resources/current_policy.yaml`, uuid.New().String(), performance_dir+"/"+plc_filename, performance_dir))
+			if err != nil {
+				klog.Exitf("Error patching policy with unique name: %s; %s", err, sed)
+			}
+
+			creation_output, err := exec.Command("oc", "apply", "-f", performance_dir+"/resources/current_policy.yaml").CombinedOutput()
+			if err != nil {
+				klog.Error("Error creating policy: %s, %s", err, string(creation_output))
+				batch_fails += 1
+			}
+		}
+
+		// sleep for the remainder of per_batch_sleep
+		end := time.Now()
+		elapsed := end.Sub(start)
+		bonus_time := time.Duration(per_batch_sleep)*time.Minute - elapsed
+
+		klog.V(2).Infof(
+			"%d policies created in %.2f seconds! Waiting an additional %.2f seconds for policies to process...\n",
+			n_per_batch-batch_fails,
+			elapsed.Seconds(),
+			bonus_time.Seconds(),
+		)
+
+		time.Sleep(bonus_time)
+
+		table_data = append(table_data, getMetrics(thanos_host, token, per_batch_sleep, total_plcs+n_per_batch))
+	}
+
+	printTable(table_data)
+
+	outpath := filepath.Join(".", performance_dir, "output")
+	err = os.MkdirAll(outpath, os.ModePerm)
+	if err != nil {
+		klog.Errorf("Error creating output directory: %s", err)
+	}
+	exportTable(table_data, performance_dir+"/output/"+output_filename)
+
+	klog.Info("Performance test completed! Cleaning up...")
+	bash(fmt.Sprintf("rm %s/resources/current_policy.yaml", performance_dir))
```

Comment:
I think it'd be cleaner if you created a temporary directory with Go and have a defered function that cleans up that directory.

https://pkg.go.dev/os#MkdirTemp
https://pkg.go.dev/os#TempDir